java/java.hints/src/org/netbeans/modules/java/hints/jdk/RegexExampleVisualPanel1.java

```diff
@@ -0,0 +1,160 @@
+                    .addGroup(layout.createSequentialGroup()
+                        .addComponent(regExLabel)
+                        .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED)
+                        .addComponent(expressionLabel)
+                        .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED, javax.swing.GroupLayout.DEFAULT_SIZE, Short.MAX_VALUE)
+                        .addComponent(NoLabel)
+                        .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED)
+                        .addComponent(numberField, javax.swing.GroupLayout.PREFERRED_SIZE, 42, javax.swing.GroupLayout.PREFERRED_SIZE))
+                    .addGroup(layout.createSequentialGroup()
+                        .addComponent(regenerateButton)
+                        .addGap(0, 0, Short.MAX_VALUE)))
+                .addContainerGap())
+        );
+        layout.setVerticalGroup(
+            layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
+            .addGroup(layout.createSequentialGroup()
+                .addContainerGap()
+                .addGroup(layout.createParallelGroup(javax.swing.GroupLayout.Alignment.BASELINE)
+                    .addComponent(regExLabel)
+                    .addComponent(expressionLabel)
+                    .addComponent(numberField, javax.swing.GroupLayout.PREFERRED_SIZE, javax.swing.GroupLayout.DEFAULT_SIZE, javax.swing.GroupLayout.PREFERRED_SIZE)
+                    .addComponent(NoLabel))
+                .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED)
+                .addComponent(exampleScrollPane, javax.swing.GroupLayout.PREFERRED_SIZE, 207, javax.swing.GroupLayout.PREFERRED_SIZE)
+                .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED)
+                .addComponent(regenerateButton)
+                .addContainerGap())
+        );
+    }// </editor-fold>//GEN-END:initComponents
+
+    private void regenerateButtonActionPerformed(java.awt.event.ActionEvent evt) {//GEN-FIRST:event_regenerateButtonActionPerformed
+        exampleTextArea.setText(getExamples());
+    }//GEN-LAST:event_regenerateButtonActionPerformed
+
+    // Variables declaration - do not modify//GEN-BEGIN:variables
+    private javax.swing.JLabel NoLabel;
+    private javax.swing.JScrollPane exampleScrollPane;
+    private javax.swing.JTextArea exampleTextArea;
+    private javax.swing.JLabel expressionLabel;
+    private javax.swing.JTextField numberField;
+    private javax.swing.JLabel regExLabel;
+    private javax.swing.JButton regenerateButton;
+    // End of variables declaration//GEN-END:variables
+
+    private String getExamples() {     
+        RegExParser r = new RegExParser(expression);        
+        RegEx p = r.parse();        
+        ExampleGenerator eg = new ExampleGenerator(p);        
+        ArrayList<String> generate = eg.generate(Integer.parseInt(numberField.getText()));        
```

Comment:
Please try/catch this _parseInt()_ method call or disable **Regenerate** button when _numberField_ contains text instead of number. Right now it silently throws exception and does nothing.

Reply:
Thanks for the input. Have made some changes.